### PR TITLE
Add changelog for 2.17.1

### PR DESCRIPTION
**Pull request by lsviben**
**What this PR does / why we need it**:
Add the changelog for the 2.17.1 release.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
 # Kubermatic 2.17
 
+## [v2.17.1](https://github.com/kubermatic/kubermatic/releases/tag/v2.17.1)
+
+### Security
+
+- Upgrade machine-controller to v1.27.8 [#7209](https://github.com/kubermatic/kubermatic/pull/7166) to address [runC vulnerability CVE-2021-30465](https://github.com/opencontainers/runc/security/advisories/GHSA-c3xm-pvg7-gh7r)
+
+### Bugfixes
+
```
Comment (irozzo-1A) [inline CHANGELOG.md:10]: I would mention this in the bugfixes: https://github.com/kubermatic/kubermatic/pull/7153

Comment (lsviben) [reply CHANGELOG.md:10]: Use optimistic lock when adding finalizers to prevent lost updates[#7153] ?

**Pull request merged**